Assets/Mirror/Runtime/INetworkConnection.cs

```diff
@@ -14,6 +14,10 @@ void RegisterHandler<T>(Action<INetworkConnection, T> handler)
 
         void RegisterHandler<T>(Action<T> handler) where T : IMessageBase, new();
 
+        void RegisterHandler<T1, T2>(Action<T1> handler) where T1 : IMessageBase where T2 : IMessageBase, new();
+
+        void RegisterHandler<T1, T2>(Action<INetworkConnection, T1> handler) where T1 : IMessageBase where T2 : IMessageBase, new();
+
```

Comment:
You would register a function in this case that returns T2

```suggestion
        void RegisterHandler<T1, T2>(Func<T1, T2> handler) where T1 : IMessageBase where T2 : IMessageBase, new();

        void RegisterHandler<T1, T2>(Func<INetworkConnection, T1, T2> handler) where T1 : IMessageBase where T2 : IMessageBase, new();

```

Reply:
fixed this and it helped me solve more of the puzzle. But now I am unsure what to do with: MessageHandler(handler). I think we need a new version of that.